### PR TITLE
sql: add test TestAlterRegionalByRowEnclosingRegionAddDrop

### PR DESCRIPTION
**Pull request by arulajmani**
See individual commits for details. 

There's more test scenarios I want to add where we sandwich an ADD/DROP region inside of an ALTER to REGIONAL BY ROW/or an operation on one of these tables, but here's a start. The error messages aren't wrapped yet, so they may appear a bit ugly in the jobs table. What this test ensures, which I think is quite valuable, is that the rollback is fine and no manual cleanup is required.

### CODE REVIEW

File: pkg/ccl/multiregionccl/region_test.go [hunk 1]
```diff
@@ -495,50 +468,220 @@ CREATE TABLE db.rbr(k INT PRIMARY KEY, v INT NOT NULL) LOCALITY REGIONAL BY ROW;
...
+		name  string
+		setup string
+	}{
+		{
+			name:  "alter-from-global",
+			setup: `CREATE TABLE db.t () LOCALITY GLOBAL`,
+		},
+		{
+			name:  "alter-from-explicit-regional",
+			setup: `CREATE TABLE db.t () LOCALITY REGIONAL IN "us-east2"`,
```
Comment (otan) [inline pkg/ccl/multiregionccl/region_test.go:554]: i'm convinced `REGIONAL BY ROW` -> `GLOBAL` / `REGIONAL BY TABLE` is broken in that it creates extraneous partitions / zone configs, but looks like we're not there yet....

**Review by ajstorm — APPROVED** [review]
:lgtm: with just one nit

> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64017#-:-MYtViclDokQACrUYRuI:b-l8ezpm)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani and @otan)

___
*[pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go, line 69 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/64017#-MYtUfte4V45bDaeNBxS:-MYtUfte4V45bDaeNBxT:bthbv5b) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e0f498214281bbf2221792763f70915b3940f6d/pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go#L69)):*
> ```Go
> // expected partitions.
> func TestingEnsureCorrectPartitioning(
> 	sqlDB *gosql.DB, expectedPartitions []string, expectedIndexes []string, tableFQN string,
> ```

Nit: for expected partitions here, does it make sense to derive that from a SHOW REGIONS FROM DATABASE command?  That would make this both easier to call, less error prone (hopefully), and a truer validation.

**Review by arulajmani — COMMENTED** [review]
TFTR! bors r=ajstorm

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64017#-:-MYuQsMhCl9Aiu2TSk4h:b-2jcdc9)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajstorm, @arulajmani, and @otan)

___
*[pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go, line 69 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/64017#-MYtUfte4V45bDaeNBxS:-MYuQrBI2BTLLrSlQZyv:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e0f498214281bbf2221792763f70915b3940f6d/pkg/ccl/multiregionccl/multiregionccltestutils/testutils.go#L69)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: for expected partitions here, does it make sense to derive that from a SHOW REGIONS FROM DATABASE command?  That would make this both easier to call, less error prone (hopefully), and a truer validation.
</blockquote></details>

Done.

**Pull request merged**